corrector_module/opmon_corrector/corrector_batch.py

```diff
@@ -91,40 +114,76 @@ def _batch_run(self, process_dict):
         m = multiprocessing.Manager()
         to_remove_queue = m.Queue()
 
-        for message_id in doc_map:
-            documents = doc_map[message_id]
+        for x_request_id in doc_map:
+            documents = doc_map[x_request_id]
             data = dict()
             data['logger_manager'] = self.logger_m
             data['document_manager'] = doc_m
-            data['message_id'] = message_id
+            data['x_request_id'] = x_request_id
             data['documents'] = documents
             data['to_remove_queue'] = to_remove_queue
             list_to_process.put(data)
             doc_len += len(documents)
 
-        # Sync
-        # time.sleep(5)
+        self._process_workers(list_to_process, duplicates)
 
-        # Create pool of workers
-        pool = []
-        for i in range(self.settings["corrector"]["thread-count"]):
-            # Configure worker
-            worker = CorrectorWorker(self.settings, f'worker_{i}')
-            p = multiprocessing.Process(target=worker.run, args=(list_to_process, duplicates))
-            pool.append(p)
+        # Go through the to_remove list and remove the duplicates
+        element_in_queue = True
+        total_raw_removed = 0
+        while element_in_queue:
+            try:
+                element = to_remove_queue.get(False)
```

Comment:
`to_remove_queue.get(False)` seems quite odd. What is `to_remove_queue`? 🤔

Reply:
`to_remove_queue` is `multiprocessing.Queue` object, which has `get()` and  `False` mean "return an item if one is immediately available, else raise the `queue.Empty` "
https://docs.python.org/3.9/library/multiprocessing.html#multiprocessing.Queue.get